### DISCUSSION TITLE
Modal set inside table row issue

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#46870] [General]
Hello, we have a custom component that uses the AntD table. Each table row renders a button that opens a modal when clicked. The rows also have an `onRow` prop with the `onClick` callback to run code when the row is clicked. I set `stop.propagation` on the button click event so the row click callback is not triggered. However, after the modal is open, if one clicks inside the modal the table row click event triggers.

**Comment by yoyo837** [comment]
Is your Modal inside a row?

**Reply by bombillazo** [reply]
Yes

**Reply by yoyo837** [reply]
Move the Modal out of the rows and keep it single. When any row needs to open a modal, open the same and single modal. The content of the modal is reactive controlled based on the row data, such as the row `id`.

**Reply by bombillazo** [reply]
This complicates our component logic since we are using a component made up of a modal and a button. That would require us to move all logic of that component outside to where the table is defined.
